test/WebJobs.Script.Tests/Diagnostics/FileTraceWriterTests.cs

```diff
@@ -31,17 +31,40 @@ public FileTraceWriterTests()
         [Fact]
         public async Task MultipleConcurrentInstances_WritesAreSerialized()
         {
+            void FlushCallback(bool flushed, object state)
+            {
+                if (flushed && state is TaskCompletionSource<object> tcs)
+                {
+                    tcs.SetResult(null);
+                }
+            }
+
+            int numLines = 100;
+
+            async Task WriteLogsWithState()
```

Comment:
Why these changes? Does this test still pass without these modifications - hard to understand what you're changing here.

Reply:
These were required as the behavior the test was relying on has changed. The test expected the flush to happen even if concurrent calls were being made. That is no longer the case and those calls will return immediately when the lock isn't acquired, so this additional logic here is needed.